src/factories/StoneFactory.js

```diff
@@ -1,20 +1,32 @@
-import { coal, stone } from '../data/ores'
-import { COAL_SPAWN_CHANCE } from '../constants'
+/** @typedef {import("../index").farmhand.item} farmhand.item */
 import { randomNumberService } from '../common/services/randomNumber'
+import { Factory } from '../interfaces/Factory'
+import { coal, saltRock, stone } from '../data/ores'
+import {
+  COAL_SPAWN_CHANCE,
+  SALT_ROCK_SPAWN_CHANCE,
+  STONE_SPAWN_CHANCE,
+} from '../constants'
 
 /**
  * Resource factory used for spawning stone
  * @constructor
  */
-export default class StoneFactory {
+export default class StoneFactory extends Factory {
   /**
    * Generate resources
-   * @returns {Array} an array of stone and coal resources
+   * @returns {Array.<farmhand.item>} an array of stone and coal resources
    */
   generate() {
     let resources = []
 
-    resources.push(this.spawnStone())
+    if (randomNumberService.isRandomNumberLessThan(STONE_SPAWN_CHANCE)) {
+      resources.push(this.spawnStone())
+    }
+
+    if (randomNumberService.isRandomNumberLessThan(SALT_ROCK_SPAWN_CHANCE)) {
+      resources.push(this.spawnSaltRock())
+    }
```

Comment:
now that we're spawning more things in here i see some room to optimize this. something like...

```js
// this can live outside of generate just in this file, it's more of a "config" i suppose
const spawnableResources = [
  [stone, STONE_SPAWN_CHANCE],
  [saltRock, SALT_ROCK_SPAWN_CHANCE],
  [coal, COAL_SPAWN_CHANCE],
]

//generate can then work like this, and there is no need for the "spawn" methods since they don't actually do anything special
generate() {
  return spawnableResources.map((item) => {
    const [resource, spawnChance] = item
  
    if (isRandomNumberLessThan(spawnChance)) {
      return resource
    }
  
    return null
  }).filter((resource) => resource)
}
```

Reply:
Great idea! I riffed on this a bit and went something slightly different: https://github.com/jeremyckahn/farmhand/commit/6f34a327920d3394bd49121ebd545252936c0d8f

I like your filter approach, I just tend to reach for syntax-level constructs (like `for...of`) rather than method calls where I can. 🙂